src/core/thread/router_table.cpp

```diff
@@ -539,39 +541,9 @@ void RouterTable::ProcessTlv(const Mle::RouteTlv &aTlv)
         UpdateAllocation();
         Get<Mle::MleRouter>().ResetAdvertiseInterval();
     }
-}
-
-void RouterTable::ProcessTlv(const ThreadRouterMaskTlv &aTlv)
-{
-    bool allocationChanged = false;
-
-    mRouterIdSequence            = aTlv.GetIdSequence();
-    mRouterIdSequenceLastUpdated = TimerMilli::GetNow();
-
-    for (uint8_t routerId = 0; routerId <= Mle::kMaxRouterId; routerId++)
-    {
-        if (aTlv.IsAssignedRouterIdSet(routerId) == IsAllocated(routerId))
-        {
-            continue;
-        }
-
-        allocationChanged = true;
-
-        if (aTlv.IsAssignedRouterIdSet(routerId))
-        {
-            mAllocatedRouterIds.Add(routerId);
-        }
-        else
-        {
-            mAllocatedRouterIds.Remove(routerId);
```

Comment:
There is a difference in behavior here where we don't do 
```
     {
            Router *router = GetRouter(routerId);

            OT_ASSERT(router != NULL);
            router->SetNextHop(Mle::kInvalidRouterId);
            RemoveRouterLink(*router);
     }
```
compared to the new common `ProcessRouterIdSet()`. 

I am not sure if it matters. I think it's fine to use common method. @jwhui , @librasungirl  do you agree?

Reply:
Good point. I think the added logic in this case is OK.